httpcore/_sync/http2.py

```diff
@@ -338,16 +360,21 @@ def _receive_events(
                             self._receive_remote_settings_change(event)
                             trace.return_value = event
 
-                    event_stream_id = getattr(event, "stream_id", 0)
-
-                    # The ConnectionTerminatedEvent applies to the entire connection,
-                    # and should be saved so it can be raised on all streams.
-                    if hasattr(event, "error_code") and event_stream_id == 0:
-                        self._connection_error_event = event
-                        raise RemoteProtocolError(event)
-
-                    if event_stream_id in self._events:
-                        self._events[event_stream_id].append(event)
+                    elif isinstance(
+                        event,
+                        (
+                            h2.events.ResponseReceived,
+                            h2.events.DataReceived,
+                            h2.events.StreamEnded,
+                            h2.events.StreamReset,
+                        ),
+                    ):
+                        event_stream_id = getattr(event, "stream_id", 0)
```

Comment:
Is `getattr` here still needed? I've checked all four event types and it seems that stream_id exists in all of them (at least it is initialised to None in the constructors).

Reply:
Good spot thanks. That's a nice bit of cleaning up there.